### PR TITLE
Support proxy setting during assembly

### PR DESCRIPTION
**Pull request by nak3**
As a s2i rust user, we would like to setup proxy for cargo
easily.

So, this patch introduces `PROXY` environment variable and adds proxy
setting*1 to `$HOME/.cargo/config` during assembly. This makes it
possible to build behind proxy.

*1 http://doc.crates.io/config.html#configuration-keys

### CODE REVIEW

**Comment by nak3** [comment]
cc @kargakis

File: README.md [hunk 1]
```diff
@@ -30,6 +30,15 @@ Finally, run the image:
...
 ```
 
+Environment variables
+---------------------
+
+Application developers can use the following environment variables to configure the runtime behavior of this image:
+
+NAME        | Description
+------------|-------------
+PROXY       | Use a cargo proxy during assembly
```
Comment (kargakis) [inline README.md]: Should this be `HTTP_PROXY` which seems like the standard way to use an HTTP proxy?

Comment (nak3) [reply README.md]: I think `PROXY` is better, as cargo's proxy setting is [libcurl format](https://curl.haxx.se/libcurl/c/CURLOPT_PROXY.html) and allows users to write it with schema not only `proxy=https://...` or `proxy=https://...`, but also `proxy=socks5h://...`.

So, in the same way, `PROXY=socks5h://...` is comfortable for the s2i-rust users,

Comment (kargakis) [reply README.md]: sgtm

**Pull request merged**